c/meterpreter/source/extensions/winpmem/winpmem.h

```diff
@@ -1,5 +1,6 @@
-#include "common.h"
-
+extern "C" {
```

Comment:
In the past I've mostly seen this metadata exist in the header files directly  `common.h` file, like:

Similar to this pattern

https://github.com/rapid7/metasploit-payloads/blob/6e08d1f9812aa4d7a76b451fd5e0bae03975bb91/c/meterpreter/source/extensions/python/Modules/config.c.in#L20-L22

Maybe @smcintyre-r7 has more insight into why we'd preference this metadata living in winpmem versus being directly inside `common.h` 👀 

Potentially it's because we don't have `__cplusplus` available in this version of visual studio to do the required `#ifdef __cplusplus` check? 🤔 

https://devblogs.microsoft.com/cppblog/msvc-now-correctly-reports-__cplusplus/